src/app_service/service/message/service.nim

```diff
@@ -416,6 +410,10 @@ QtObject:
       var receivedData = DiscordChannelImportFinishedSignal(e)
       self.resetMessageCursor(receivedData.channelId)
       self.asyncLoadMoreMessagesForChat(receivedData.channelId)
+    
+    self.events.on(SIGNAL_CHAT_LEFT) do(e: Args):
+      var chatArg = ChatArgs(e)
+      self.resetMessageCursor(chatArg.chatId)
```

Comment:
This is to make sure that that we reset the cursor, so that if we reopen the chat, we reload from the "start" (newest messages). Also, it deletes the cursor from the table, so it's lighter.